web/src/engine/websites/decorators/Common.ts

```diff
@@ -450,10 +449,12 @@ export function PagesSinglePageJS(script: string, delay = 0) {
  * @param priority - The importance level for ordering the request for the image data within the internal task pool
  * @param signal - An abort signal that can be used to cancel the request for the image data
  * @param detectMimeType - Force a fingerprint check of the image data to detect its mime-type (instead of relying on the Content-Type header)
+ * @param deProxifyLink - Remove common image proxies (default false)
  */
-export async function FetchImageAjax(this: MangaScraper, page: Page, priority: Priority, signal: AbortSignal, detectMimeType = false): Promise<Blob> {
+export async function FetchImageAjax(this: MangaScraper, page: Page, priority: Priority, signal: AbortSignal, detectMimeType = false, deProxifyLink = false): Promise<Blob> {
     return this.imageTaskPool.Add(async () => {
-        const request = new FetchRequest(page.Link.href, {
+        const imageLink = deProxifyLink ? DeProxify(page.Link) : page.Link;
+        const request = new FetchRequest(imageLink.href, {
             signal: signal,
             headers: {
                 Referer: page.Parameters?.Referer || page.Link.origin,
```

Comment:
Maybe use `imageLink.origin` instead of `page.Link.origin`?
Probably depends what the image proxy would set as the `Referer` header.

Reply:
Hum. Seeing that we already deproxify pictures links from all madara and mangastream websites before that change,(meaning they have no referrer problem at all), and that only a few websites will use deproxify right here i'll make that change.